src/AaveV3PayloadBase.sol

```diff
@@ -0,0 +1,161 @@
+ *   - Updates of price feeds
+ *   - Updates of interest rate strategies.
+ *   - Updates of borrow parameters (flashloanable, stableRateModeEnabled, borrowableInIsolation, withSiloedBorrowing, reserveFactor)
+ *   - Updates of collateral parameters (ltv, liq threshold, liq bonus, liq protocol fee, debt ceiling)
+ * @author BGD Labs
+ */
+abstract contract AaveV3PayloadBase {
+  using Address for address;
+
+  IEngine public immutable LISTING_ENGINE;
+
+  constructor(IEngine engine) {
+    LISTING_ENGINE = engine;
+  }
+
+  /// @dev to be overriden on the child if any extra logic is needed pre-listing
+  function _preExecute() internal virtual {}
+
+  /// @dev to be overriden on the child if any extra logic is needed post-listing
+  function _postExecute() internal virtual {}
+
+  function execute() external {
+    _preExecute();
+
+    IEngine.Listing[] memory listings = newListings();
+    IEngine.ListingWithCustomImpl[] memory listingsCustom = newListingsCustom();
+    IEngine.CapsUpdate[] memory caps = capsUpdates();
+    IEngine.CollateralUpdate[] memory collaterals = collateralsUpdates();
+    IEngine.BorrowUpdate[] memory borrows = borrowsUpdates();
+    IEngine.PriceFeedUpdate[] memory priceFeeds = priceFeedsUpdates();
+    IEngine.RateStrategyUpdate[] memory rates = rateStrategiesUpdates();
+
+    if (listings.length != 0) {
+      address(LISTING_ENGINE).functionDelegateCall(
+        abi.encodeWithSelector(LISTING_ENGINE.listAssets.selector, getPoolContext(), listings)
+      );
+    }
+
+    if (listingsCustom.length != 0) {
+      address(LISTING_ENGINE).functionDelegateCall(
+        abi.encodeWithSelector(
+          LISTING_ENGINE.listAssetsCustom.selector,
+          getPoolContext(),
+          listingsCustom
+        )
+      );
+    }
+
+    if (borrows.length != 0) {
+      address(LISTING_ENGINE).functionDelegateCall(
+        abi.encodeWithSelector(LISTING_ENGINE.updateBorrowSide.selector, borrows)
+      );
+    }
+
+    if (collaterals.length != 0) {
+      address(LISTING_ENGINE).functionDelegateCall(
+        abi.encodeWithSelector(LISTING_ENGINE.updateCollateralSide.selector, collaterals)
+      );
+    }
+
+    if (rates.length != 0) {
+      address(LISTING_ENGINE).functionDelegateCall(
+        abi.encodeWithSelector(LISTING_ENGINE.updateRateStrategies.selector, rates)
+      );
+    }
+
+    if (priceFeeds.length != 0) {
+      address(LISTING_ENGINE).functionDelegateCall(
+        abi.encodeWithSelector(LISTING_ENGINE.updatePriceFeeds.selector, priceFeeds)
+      );
+    }
+
+    if (caps.length != 0) {
+      address(LISTING_ENGINE).functionDelegateCall(
+        abi.encodeWithSelector(LISTING_ENGINE.updateCaps.selector, caps)
+      );
+    }
+
+    _postExecute();
+  }
+
+  /** @dev Converts basis points to RAY units
+   * e.g. 10_00 (10.00%) will return 100000000000000000000000000
+   */
+  function _bpsToRay(uint256 amount) internal pure returns (uint256) {
+    return (amount * 1e27 / 10_000);
```

Comment:
The brackets changed here from the original. They should either be removed or left in the original structure.

Reply:
Yep thanks fixed